src/JuNix/JuNix.jl

```diff
@@ -147,34 +147,36 @@ end
 
 function write_depot(
     depot::Dict{String,Fetcher},
-    meta::Dict{String,Any},
     opts::Options,
     package_path::String,
     out_path::String,
     name::String,
 )
     io = IOBuffer(; append = true)
-    write(io, "{ pkgs ? import <nixpkgs> {} }: {\n")
-    write(io, "meta = ")
-    Nix.print(io, meta)
-    write(io, ";\n")
-    write(io, "depot = {")
     for path in sort(collect(keys(depot)))
-        Nix.print(io, path)
-        write(io, " = (")
-        Nix.print(io, depot[path])
-        write(io, ");\n")
+        write(io, "[\"", depot[path].args["name"], "\"]\n")
+        TOML.print(io, Dict("path" => path))
+        TOML.print(io, depot[path].args)
+        write(io, "\n")
     end
-    write(io, "};\n")
-    write(io, '}')
 
-    depotfile_path = normpath(joinpath(package_path, "Depot.nix"))
+    toml = TOML.parse(io)
+    for path in sort(collect(keys(toml)))
+        pop!(toml[path], "name")
+    end
+
+    arch = if "x86_64" in opts.arch "x86_64" else "aarch64" end
+    os = if "linux" in opts.os "linux" else "darwin" end
```

Comment:
Is there a way to get the string from opts?

Reply:
@johnnychen94 Could you give this guy some help here?